csrc/python_frontend/python_bindings.cpp

```diff
@@ -593,6 +593,10 @@ void initNvFuserPythonBindings(PyObject* module) {
           "_fusion_ir",
           [](FusionDefinition& self) { return self.fusionIr(); },
           py::return_value_policy::reference)
+      .def(
+          "_user_schedule_ir",
```

Comment:
IIUC, userScheduleIr can be safely called after a single execution with a custom schedule.

But we can also safely call it outside of `def schedule(self):` as well, my question is, is that an intended usage? Otherwise, if we are only supposed to use it inside `schedule`, we might as well keep this method under `nvf_sched` instead?

I'm not pushing strongly against this, but if we decide to keep the API as-is, I'd highly recommend relaxing the NVF_CHECK above with proper warning/recommendation on proper usage.

Reply:
It could be useful to print fusion IR outside of the `schedule` function. I'll keep it and replace the NVF_CHECK with a string message.

Reply:
I moved the method to `nvf_sched` since `user_sched_` is reset after `finalize_schedule` is called and is not available outside of `def schedule`.